usb_protocol/emitters/descriptors/hid.py

```diff
@@ -0,0 +1,87 @@
+import unittest
+
+from contextlib import contextmanager
+
+from ..           import emitter_for_format
+from ..descriptor import ComplexDescriptorEmitter
+from ...types.descriptors.hid import \
+    HIDDescriptor as HIDDescriptorType
+from ...types.descriptors.hid import *
+
+ReportDescriptorEmitter = emitter_for_format(ReportDescriptor)
+
+_hid_item_length = [ 0, 1, 2, 4 ]
+
+class HIDDescriptor(ComplexDescriptorEmitter):
+    DESCRIPTOR_FORMAT = HIDDescriptorType
+
+    def add_report(self, report_enum, *report_data):
```

Comment:
Where does report come from here? To me, when I talk about HID reports, they are the reports exchanged between the host and the device. This is actually a HID report descriptor item, I would rename it `add_item`. I would also rename `add_input` to `add_item_item`, etc. but that's just nitpicking.

Reply:
I do suppose this can be documented better. A HID report item consists of a 4 bit tag and a 2 bit type, however, tags are not quite consistent within types, so only the 6 bit combined tag/type value has any map-able meaning. Input/output/feature items are special, because while other report items pretty consistently store byte-oriented data within their additional data fields, input/output/feature items contain a 9-bit bitmap of flags (only 8 bits of which are supported within this PR, as setting the 9th bit will make things a bit of a mess). Input/output/feature items have convenience functions for this reason, as generating them is less trivial than other report items.

Reply:
Sorry for not explaining correctly. My point was not about the documentation, although that is always better, but about the naming of the method. I believe the name `add_report` is not correct, IMO it should be called `add_item` instead :blush:

Reply:
Do the changes in 2e7f95f address this?

Reply:
Ah sorry, missed that. Yes!

Reply:
I would also rename `add_report_raw` acordingly, but overall is fine.